src/test/test-capability.c

```diff
@@ -270,7 +271,8 @@ static void test_capability_get_ambient(void) {
 
                 assert_se(capability_get_ambient(&c) >= 0);
                 assert_se(x < 0 || FLAGS_SET(c, UINT64_C(1) << CAP_MKNOD));
-                assert_se(x < 0 || FLAGS_SET(c, UINT64_C(1) << CAP_LINUX_IMMUTABLE));
+                if (detect_container() <= 0) /* Not available in Docker */
```

Comment:
uh, this is misleading. docker gives you control of this iirc. the problem here is that the cap bounding set doesn't have these caps from the start, and that's what we should check for.

Reply:
simply skip the whole test if `prctl(PR_CAPBSET_READ, CAP_MKNOD)` or `prctl(PR_CAPBSET_READ, CAP_LINUX_IMMUTABLE)` return zero.

```c
r = prctl(PR_CAPBSET_READ, CAP_MKNOD);
assert_se(r >= 0);
if (r == 0)
        return;
```

and the same for CAP_LINUX_IMMUTABLE.